app/models/piece.rb

```diff
@@ -18,6 +18,38 @@ def capture_piece(x, y)
       fail "There is no piece of the opposite color to capture."
     end
   end
+
+  def move_to(destination_x, destination_y)
+    if self.valid_move?(destination_x, destination_y)
+      if self.moves < 2
+        moves_updated = self.moves + 1
+        self.update_attributes(:x_position => destination_x, :y_position => destination_y, :moves => moves_updated)
+      else
+        self.update_attributes(:x_position => destination_x, :y_position => destination_y)
+      end
+      if self.game.is_piece_present?(destination_x, destination_y) && 
+        self.game.get_piece(destination_x, destination_y).get_color != self.get_color
+        self.capture_piece(destination_x, destination_y)
+      else
+        self.en_passant(destination_x, destination_y)
+      end
+    end
+  end
+
+  def en_passant(destination_x, destination_y)
+    if destination_y == 2 || destination_y == 5
+      destination_y == 2 ? incrementer = 1 : incrementer = -1
+      incremented_y = destination_y + incrementer
+      if self.game.is_piece_present?(destination_x, incremented_y)
+        other_piece = game.get_piece(destination_x, incremented_y)
+        if other_piece.type == 'Pawn' && 
+          other_piece.moves == 1 && 
```

Comment:
Each piece has a moved column that counts how many moves that piece has made (up to 2 because that is all we care about).